ticker/simpleMock.go

```diff
@@ -0,0 +1,90 @@
+package ticker
+
+import (
+	"errors"
+	"sync/atomic"
+	"time"
+)
+
+var (
+	// ErrTimeout happens when the force tick cannot be delivered in time.
+	ErrTimeout = errors.New("force tick timeout")
+
+	// ErrPaused happens when the ticker is paused and the caller still
+	// wants to force a tick.
+	ErrPaused = errors.New("cannot force feed because ticker is paused")
+)
+
+// SimpleMock implements the Ticker interface, and provides a method of
+// force-feeding ticks, even while paused.
+type SimpleMock struct {
```

Comment:
I don't think we should add yet another ticker mock. Instead we can
1. Modify the existing mock to not tick automatically.
2. I think you can use the existing one by creating it, stopping it, then force ticking as you see fit.

Reply:
I can't stop it in the test, because start/stop is something that is controlled from the tested code. Whether it is running is something that can be asserted. At a quick glance, I think the unit tests that use the existing Mock (`link_test.go`) may not be optimal. I see quite some `time.Sleep` in there. If you mock the Ticker properly, this shouldn't be necessary.

I think I'd rather move the exiting Mock into `link_test`, to discourage people from using this hybrid time/force construction in unit tests.

Reply:
SimpleMock removed completely in favor of time.Timer.